src/OrbitGl/CaptureWindowTest.cpp

```diff
@@ -79,4 +81,193 @@ TEST_F(UnitTestCaptureWindow, SlidersRespondToMouseLeave) {
+        EXPECT_GT(time_graph_->GetMinTimeUs() * 1000, 0);
+        break;
+      case PosWithinCapture::kInBetween:
+        EXPECT_GT(slider_->GetPosRatio(), 0.0);
+        EXPECT_LT(time_graph_->GetMaxTimeUs() * 1000, time_graph_->GetCaptureMax());
+        EXPECT_GT(time_graph_->GetMinTimeUs() * 1000, 0);
+        break;
+    }
+  }
+
+ private:
+  CaptureClientAppInterfaceFake capture_client_app_fake_;
+  std::unique_ptr<orbit_client_data::CaptureData> capture_data_;
+
+  void CreateSchedulerTimers() {
+    using orbit_client_protos::TimerInfo;
+    TimerInfo timer;
+    timer.set_start(0);
+    timer.set_end(100);
+    timer.set_thread_id(TrackTestData::kThreadId);
+    timer.set_processor(0);
+    timer.set_depth(0);
+    timer.set_type(TimerInfo::kCoreActivity);
+    time_graph_->ProcessTimer(timer, nullptr);
+  }
+};
+
+TEST_F(NavigationTestCaptureWindow, ZoomTimeWorksAsExpected) {
+  ExpectInitialState();
+
+  const Vec2i kTimeGraphSize = viewport_.WorldToScreen(time_graph_->GetSize());
+
+  // Zoom time: Zoom in and out horizontally, assure we've zoomed in, and zoomed out
+  // again
+  // Test 1: Middle of the capture window
+  int x = kTimeGraphSize[0] / 2;
+  int y = kTimeGraphSize[1] - 5;
+  MouseWheelMoved(x, y, 1, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kInBetween);
+  MouseWheelMoved(x, y, -1, false);
+  ExpectInitialState();
+
+  // Keyboard
+  MouseMoved(x, y, false, false, false);
+  KeyPressed('W', false, false, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kInBetween);
+  KeyPressed('S', false, false, false);
+  ExpectInitialState();
+
+  // Test 2: Left of the capture window
+  x = 0;
+  y = kTimeGraphSize[1] - 5;
+  MouseWheelMoved(x, y, 1, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kLeft);
+  MouseWheelMoved(x, y, -1, false);
+  ExpectInitialState();
+
+  // Keyboard
+  MouseMoved(x, y, false, false, false);
+  KeyPressed('W', false, false, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kLeft);
+  KeyPressed('S', false, false, false);
+  ExpectInitialState();
+
+  // Test 3: Right of time graph
+  x = kTimeGraphSize[0];
+  y = kTimeGraphSize[1] - 5;
+  MouseWheelMoved(x, y, 1, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kRight);
+  MouseWheelMoved(x, y, -1, false);
+  ExpectInitialState();
+
+  // Keyboard
+  MouseMoved(x, y, false, false, false);
+  KeyPressed('W', false, false, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kRight);
+  KeyPressed('S', false, false, false);
+  ExpectInitialState();
+}
+
+TEST_F(NavigationTestCaptureWindow, VerticalZoomWorksAsExpected) {
+  ExpectInitialState();
+
+  const Vec2i kTimeGraphSize = viewport_.WorldToScreen(time_graph_->GetSize());
+  int x = kTimeGraphSize[0] / 2;
+  int y = kTimeGraphSize[1] - 5;
+
+  float old_height = time_graph_->GetTrackContainer()->GetVisibleTracksTotalHeight();
+  MouseWheelMoved(x, y, 1, true);
+  UpdateChildrenPosAndSize();
```

Comment:
I am a bit surprised that we need to call this here explicitly (shouldn't that be done by the `MouseWheelMoved` call (or something that wraps those two)?

Probably not for this PR though.

Reply:
I'm planning to remove this method as part of moving the sliders to be TimeGraph's children. Anyway I'm interested in Florian's answer.

Reply:
Sounds legit. But then, as a reader, I am wondering why those calls doe not appear in the test above.

Reply:
(now knowing that they do, but are part of the Expect... calls). I'd rather like to see this consistent.

Reply:
I ended up changing the logic in CaptureWindow a bit, I think it's now easier to understand and matches this idea of Update / Draw better, PTAL @karupayun and @florian-kuebler - the "Update" part has now moved to `CaptureWindow::PreRender()`, this is executed in every frame and will determine if a redraw is needed.

I may split this into a separate PR if you agree...